tutorials/t8_step3_1_adapt_forest_pras.cxx

```diff
@@ -0,0 +1,325 @@
+  t8_cmesh_t          cmesh;
+  t8_forest_t         forest;
+  /* The prefix for our output files. */
+  const char         *prefix_uniform = "t8_step3_1_uniform_forest";
+        char         *prefix_adapt = "t8_step3_1_adapted_forest";
+  /* The uniform refinement level of the forest. */
+  const int           level = 3;
+
+  /* Initialize MPI. This has to happen before we initialize sc or t8code. */
+  mpiret = sc_MPI_Init (&argc, &argv);
+  /* Error check the MPI return value. */
+  SC_CHECK_MPI (mpiret);
+
+  /* Initialize the sc library, has to happen before we initialize t8code. */
+  sc_init (sc_MPI_COMM_WORLD, 1, 1, NULL, SC_LP_ESSENTIAL);
+  /* Initialize t8code with log level SC_LP_PRODUCTION. See sc.h for more info on the log levels. */
+  t8_init (SC_LP_PRODUCTION);
+
+  /* Print a message on the root process. */
+  t8_global_productionf (" [step3] \n");
+  t8_global_productionf
+    (" [step3] Hello, this is the step3 example of t8code.\n");
+  t8_global_productionf
+    (" [step3] In this example we will refine and coarsen a forest.\n");
+  t8_global_productionf (" [step3] \n");
+
+  /* We will use MPI_COMM_WORLD as a communicator. */
+  comm = sc_MPI_COMM_WORLD;
+
+  /*
+   * Setup.
+   * Build cmesh and uniform forest.
+   */
+
+  /* Build a cube cmesh with tet, hex, and prism trees. */
+  cmesh = t8_cmesh_new_hypercube_hybrid (comm, 0, 0);
+  t8_global_productionf (" [step3] Created coarse mesh.\n");
+  forest =
+    t8_forest_new_uniform (cmesh, t8_scheme_new_default_cxx (), level, 0,
+                           comm);
+
+  /* Print information of the forest. */
+  t8_global_productionf (" [step3] Created uniform forest.\n");
+  t8_global_productionf (" [step3] Refinement level:\t%i\n", level);
+  t8_step3_print_forest_information (forest);
+
+  /* Write forest to vtu files. */
+  t8_forest_write_vtk (forest, prefix_uniform);
+  t8_global_productionf (" [step3] Wrote uniform forest to vtu files: %s*\n",
+                         prefix_uniform);
+
+  /*
+   *  Adapt the forest.
+   */
+
+  /* Adapt the forest. We can reuse the forest variable, since the new adapted
+   * forest will take ownership of the old forest and destroy it.
+   * Note that the adapted forest is a new forest, though. 
+   *
+   * !!!
+   * Alteration for 3_1 task of 10 timesteps and moving refinement sphere.
+   * for-loop runs 10 times with the uniform cube as input
+   */
+     
+  for ( counter = 0.0; counter < 1.1; counter = counter +  0.1)
+  {    
+
+	// Adapting the forest, beginning with a uniform cube for each timestep          
+  	forest = t8_step3_adapt_forest (t8_forest_new_uniform (cmesh, t8_scheme_new_default_cxx (), level, 0,comm));
+
+	  /*
+	   *  Output.
+	   */
+
+	  // A quick way of naming the output files done as char array with the counter in it
+	  char array[20];
+          sprintf(array, "%f", counter);
+
+	  /* char         *prefix_adapt = "t8_step3_1_adapted_forest"; */
+	  
+	  char         *prefix_adapt = array;
```

Comment:
I think you don't have to use the `prefix_adapt` variable. You can directly pass your `array` variable around. In C/C++ it is also essentially a pointer to a `char` array.

Reply:
Of course, `array` as variable name is not a good coding style. ;)